jupyter_server/gateway/gateway_client.py

```diff
@@ -353,12 +371,18 @@ def init_static_args(self):
         perform this operation once.
 
         """
-        # Ensure that request timeout and KERNEL_LAUNCH_TIMEOUT are the same, taking the
-        #  greater value of the two.
-        if self.request_timeout < float(GatewayClient.KERNEL_LAUNCH_TIMEOUT):
-            self.request_timeout = float(GatewayClient.KERNEL_LAUNCH_TIMEOUT)
-        elif self.request_timeout > float(GatewayClient.KERNEL_LAUNCH_TIMEOUT):
-            GatewayClient.KERNEL_LAUNCH_TIMEOUT = int(self.request_timeout)
+        # Ensure that request timeout and KERNEL_LAUNCH_TIMEOUT are in sync, taking the
+        #  greater value of the two and taking into account the following relation:
+        #  request_timeout = KERNEL_LAUNCH_TIME + padding
+        minimum_request_timeout = (
+            float(GatewayClient.KERNEL_LAUNCH_TIMEOUT) + self.kernel_launch_timeout_pad
+        )
+        if self.request_timeout < minimum_request_timeout:
+            self.request_timeout = minimum_request_timeout
+        elif self.request_timeout > minimum_request_timeout:
+            GatewayClient.KERNEL_LAUNCH_TIMEOUT = (
+                int(self.request_timeout) - self.kernel_launch_timeout_pad
+            )
```

Comment:
Is the `elif` necessary?  Just thinking that users will drive this using KLT and if `request_timeout` is significantly greater than KLT, does that matter so long as it's at least >= KLT + pad.  I'm just not sure if automatically increasing KLT will introduce some additional side effects during the launch that otherwise weren't anticipated.

Reply:
I agree to not touch KERNEL_LAUNCH_TIMEOUT in case `request_timeout` is significantly greater than KLT, the only problem I see is breaking up the previous behavior:
- by just setting request_timeout, kernel launch time would be adjusted too, now this would not happen anymore.

Definitely the code would become simpler, as we just increase request_timeout to ensure the minimum padding.

If you think backwards compatibility is not a problem, I opt for what you proposed